wordpress-mobile/WordPress-iOS
# [Gutenberg] Sync editor setting per site

This PR is a work-in-progress, targeting a feature branch to implement https://github.com/wordpress-mobile/gutenberg-mobile/issues/1233

This PR adds `mobileEditor` and `webEditor` to the Blog model, and are synchronized with the server.

Related `WordPressKit` PR: https://github.com/wordpress-mobile/WordPressKit-iOS/pull/166 

The editor selection setting is now per site, but the switch is still App-wide, so (just for now), changing the `Use Block Editor` switch will iterate through all the user's blogs and set the appropriate value. This will change when we merge https://github.com/wordpress-mobile/WordPress-iOS/pull/12166, that move the switch UI to the Site Settings section.

**Note:** As mentioned on https://github.com/wordpress-mobile/gutenberg-mobile/issues/1233:
> Check the values of the web and mobile preferences. If both are set to gutenberg launch GB for new post, otherwise Aztec.

This means that switching the `Gutenberg Editor` ON, won't necessarily open gutenberg unless the web setting is also set to gutenberg.

To test:
- Check that the editor setting get's sync from remote when the app lunches or a site is selected.
- Check that the proper value is stored when switching the `Use Block Editor` switch.

Update release notes:
